src/x86/nt.rs

```diff
@@ -0,0 +1,414 @@
+//! Non-temporal store operations.
+use core::{marker::PhantomData, ptr};
+
+#[cfg(target_arch = "x86")]
+use core::arch::x86::{
+    self as arch, __m128, __m128d, __m128i, __m256, __m256d, __m256i, _mm_sfence,
+};
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64::{
+    self as arch, __m128, __m128d, __m128i, __m256, __m256d, __m256i, _mm_sfence,
+};
+
+/// Load from a 32-bit aligned address with non-temporal hint, avoiding filling the cache.
+#[inline]
+#[cfg(any())]
```

Comment:
```suggestion
```
There are a few of these

Reply:
Those should be all on feature `sse4a`, which is still unstable.

Reply:
I only found out about the SSE4a extension from this PR, didn't know that was still unstable (but also wondering if it'll ever be stable?).

It definitely makes sense to add in another PR. I think a catch-all `nightly` feature for unstableness would be fine. Do you think it should be any more granular than that?

Reply:
Adding an unstable `nightly` feature in #7

Also added a doc test job with `+avx` after that doctest scare. Hypothesizing that Miri didn't fail because `cfg!(target_feature = "avx")` was false.